src/bin/nydusd/fs_cache.rs

```diff
@@ -538,11 +573,26 @@ impl FsCacheHandler {
                     }
                     Ok(md) => {
                         let cache_file = unsafe { File::from_raw_fd(msg.fd as RawFd) };
+                        let bootstrap_fd = f.as_raw_fd();
+                        let cachefile_fd = cache_file.as_raw_fd();
                         let object = FsCacheObject::Bootstrap(Arc::new(FsCacheBootStrap {
                             bootstrap_file: f,
                             cache_file,
                         }));
                         e.insert((object, msg.fd));
+                        let len = md.len() as usize;
+                        ASYNC_RUNTIME.spawn_blocking(move || {
+                            //add slight delay to let copen reply first
+                            thread::sleep(time::Duration::from_millis(10));
+                            for _i in 0..3 {
+                                if let Ok(()) =
+                                    Self::fill_bootstrap_cache(bootstrap_fd, cachefile_fd, len)
+                                {
+                                    break;
+                                }
+                                thread::sleep(time::Duration::from_secs(2));
```

Comment:
Why do we need sleep for another 2 secs?

Reply:
just add a bit delay before retry fill bootstrap , when we get io err.